README.md

```diff
@@ -1,37 +1,129 @@
+
+async function queryDatabaseWithIterator() {
+  // create new database in memory
+  const db = new DuckDB();
+  // create a new connection to the database
+ const connection = new Connection(db);
+
+ // perform some queries
+ await connection.executeIterator("CREATE TABLE people(id INTEGER, name VARCHAR);");
+ await connection.executeIterator("INSERT INTO people VALUES (1, 'Mark'), (2, 'Hannes'), (3, 'Bob');");
+ const result = await connection.executeIterator("SELECT * FROM people;");
+
+ // fetch and print result
+ console.log(result.fetchAllRows());
 
-First build:
+ // release resources
+ connection.close();
+ db.close();
+}
 
-1. `yarn install` - installs dependencies including downloading duckdb
-2. `yarn build:ts` - builds typescript
-3. `yarn test` - runs all tests
+queryDatabaseWithIterator();
 
-Other useful scripts:
+```
 
-- `yarn build` - build everything
-- `yarn build:addon` - build just the bindings code
-- `yarn build:duckdb` - build just the duckdb database
-- `yarn build:ts` - build just the typescript code
-- `yarn lint` - lint the project
-- `yarn test` - run all tests
-- `yarn test csv` - run just the csv test suite
+#### Streaming API example
 
-## Publishing
+A demo of reading from DuckDB, transforming to CSV and writing to file using the streaming API:
+
+```
+import { Connection, DuckDB, RowResultFormat } from "node-duckdb";
+import { createWriteStream } from "fs";
+import { Transform } from "stream";
+
+
+class ArrayToCsvTransform extends Transform {
+    constructor() {
+        super({objectMode: true})
+    }
+    _transform(chunk: any[], _encoding: string, callback: any) {
+        this.push(chunk.join(",") + '\n');
+        callback();
+    }
+}
+
+async function outputToFileAsCsv() {
+    // create new database in memory
+    const db = new DuckDB();
+    // create a new connection to the database
+    // note you can execute only one streaming query at a time per one connection
+    const connection = new Connection(db);
+
+    // by default a query is a streaming one, as opposed to being materialized
+    await connection.execute("CREATE TABLE people(id INTEGER, name VARCHAR);");
+    await connection.execute("INSERT INTO people VALUES (1, 'Mark'), (2, 'Hannes'), (3, 'Bob');");
+
+    // result is a stream of arrays
+    const resultStream = await connection.execute("SELECT * FROM people;", {rowResultFormat: RowResultFormat.Array});
+
+    const transformToCsvStream = new ArrayToCsvTransform();
+    const writeStream = createWriteStream("my-people-output");
+    // objects -> csv strings -> file
+    resultStream.pipe(transformToCsvStream).pipe(writeStream);
+}
+
+outputToFileAsCsv();
+
+```
+
+#### Complete sample project
+
+You can see a complete sample project using node-duckdb [here](https://github.com/deepcrawl/node-duckdb/tree/master/examples).
+
+## API
+
+API documentation is found [here](https://github.com/deepcrawl/node-duckdb/docs/api/node-duckdb.md).
```

Comment:
Would a relative link work here? Otherwise it is hard to browse on a branch.

Reply:
They would but only in github (npm would give broken links)

Reply:
Ah, got it.